store/resource.js

```diff
@@ -45,7 +45,8 @@ class Resource {
       return request(this.url).then(
         ({ data }) => {
           this.extend(data)
-          return this.retrieve(...scopes)
+          if (scopes.length) return this.retrieve(...scopes)
```

Comment:
@Joozty the first option is to call `Resource.prototype.retrieve(...scopes)` explicitly. The second option is tweaking it a bit in the `DataProvider.prototype.retrieve`. Let me have a look a this.

However, on the one hand, this recursion is not obvious what is bad. On the other hand, it's quite nice, to my mind, because I implemented it. @Joozty what do you suggest here as a user of this class?

Reply:
I already expressed my opinion about it while again. In the nutshell, it took me a while to understand how it works but once I got it I liked it.